tests/test_core.py

```diff
@@ -0,0 +1,132 @@
+"""Tests UAP-Python using the UAP-core test suite
+"""
+import contextlib
+import dataclasses
+import logging
+import pathlib
+import platform
+from operator import attrgetter
+
+import pytest  # type: ignore
+
+if platform.python_implementation() == "PyPy":
+    from yaml import load, SafeLoader
+else:
+    try:
+        from yaml import load, CSafeLoader as SafeLoader  # type: ignore
+    except ImportError:
+        logging.getLogger(__name__).warning(
+            "PyYaml C extension not available to run tests, this will result "
+            "in dramatic tests slowdown."
+        )
+        from yaml import load, SafeLoader
+
+from ua_parser import (
+    BasicParser,
+    UserAgent,
+    OS,
+    Device,
+    ParseResult,
+    UserAgentMatcher,
+    load_builtins,
+)
+
+CORE_DIR = (pathlib.Path(__name__).parent.parent / "uap-core").resolve()
+
+
+PARSERS = [
+    pytest.param(BasicParser(load_builtins()), id="basic"),
+]
+with contextlib.suppress(ImportError):
+    from ua_parser import re2
+
+    PARSERS.append(pytest.param(re2.Parser(load_builtins()), id="re2"))
+
+UA_FIELDS = {f.name for f in dataclasses.fields(UserAgent)}
+
+
+@pytest.mark.parametrize("parser", PARSERS)
+@pytest.mark.parametrize(
+    "test_file",
+    [
+        CORE_DIR / "tests" / "test_ua.yaml",
+        CORE_DIR / "test_resources" / "firefox_user_agent_strings.yaml",
+        CORE_DIR / "test_resources" / "pgts_browser_list.yaml",
+    ],
+    ids=attrgetter("name"),
+)
```

Comment:
Originally wanted each test case to be a parametric test, however:

- yaml parsing causes the test suite to take ages to import as the yaml has to be parsed at module import
- the test suite becomes humongous as there are thousands of cases in the yaml test files
- it significantly increases pytest overhead, as pytest now needs to perform setup, run, teardown for each of them (even if the setup and teardown are trivial)
- the test ids are horrifying 😢